client/state/notices/reducer.js

```diff
@@ -31,6 +32,16 @@ export const items = createReducer( {}, {
 
 		return omit( state, noticeId );
 	},
+	[ NOTICE_CLEAR ]: ( state ) => {
+		return reduce( state, ( memo, notice, noticeId ) => {
```

Comment:
I think we need clarification on what "persistence" means in relation to notices. My understanding is that it's very much tied to the routing action, which is why it's considered in the `ROUTE_SET` action handler. I'm not sure the same holds true when clearing all notices.

Looking again at the code changes, I'm wondering if `clearNotices` as called from the form submission handler is a little heavy-handed. It seems like we should be targeting whatever previous failure/success notice had been shown specifically for the form, not clearing potentially unrelated notices.